test/functional/test_framework/evm_key_pair.py

```diff
@@ -0,0 +1,31 @@
+from eth_account import Account
+from web3 import Web3
+
+
+def validate_keys(pkey, pkey_address):
+    account = Account.from_key(pkey)
+
+    if account.address != pkey_address:
+        raise RuntimeError(
+            f"""
+            Private key does not correspond to provided address.
+            Address provided: {pkey_address}
+            Address computed: {account.address}
+            """)
+    else:
+        return pkey, pkey_address
+
+
+class KeyPair:
+    def __init__(self, node=None, pkey: str = None, pkey_address: str = None):
```

Comment:
Let keypair not take `Node`.  Keep key pair free of node dependency. 
Instead, use static `from/with` helpers for cleaner construction. 

Eg: `generate_from_node` that uses the node to construct with the right init